setup.py

```diff
@@ -1,308 +1,114 @@
-            if not static and not windows:
-                self.extensions[0].runtime_library_dirs += [os.path.join(adir, lib_path)]
-
-        # Add the static linking options.
-
-        # Add MSVC specific options.
-        if windows and not msys2:
-            self.extensions[0].extra_link_args.append('/MANIFEST')
-            self.extensions[0].define_macros.append(("MPIR", 1))
-            if not static:
-                self.extensions[0].define_macros.append(("MSC_USE_DLL", None))
-        else:
-            if mpir:
-                if static and mp_found:
-                    self.extensions[0].extra_objects.append(os.path.join(mp_found, lib_path, 'libmpir.a'))
-            else:
-                if static and mp_found:
-                    self.extensions[0].extra_objects.append(os.path.join(mp_found, lib_path, 'libgmp.a'))
-            if static and mpfr_found:
-                self.extensions[0].extra_objects.append(os.path.join(mpfr_found, lib_path, 'libmpfr.a'))
-            if static and mpc_found:
-                self.extensions[0].extra_objects.append(os.path.join(mpc_found, lib_path, 'libmpc.a'))
-
+        self.fast = False
+        self.gcov = False
+        self.vector = False
+        self.mpir = False
+        self.static = False
+        self.gdb = False
 
     def finalize_options(self):
         build_ext.finalize_options(self)
-        gmpy_build_ext.doit(self)
-
-# custom install data in order that data_files
-# get installed together with the .so
-
-class gmpy_install_data(install_data):
-    def finalize_options(self):
-        install_data.finalize_options(self)
-        install = self.distribution.get_command_obj('install')
-        self.install_dir = install.install_purelib
-
-# prepare the extension for building
-
-my_commands = {'clean' : gmpy_clean, 'build_ext' : gmpy_build_ext, 'install_data' : gmpy_install_data}
-
-gmpy2_ext = Extension('gmpy2',
-                      sources=[os.path.join('src', 'gmpy2.c')],
-                      include_dirs=['./src'],
-                      define_macros = defines)
-
-setup(name = "gmpy2",
-      version = "2.1.0a1",
-      author = "Case Van Horsen",
-      author_email = "casevh@gmail.com",
-      license = "LGPL-3.0+",
-      url = "https://github.com/aleaxit/gmpy",
-      description = "gmpy2 interface to GMP/MPIR, MPFR, and MPC for Python 2.6+ and 3.4+",
-      data_files = [('', ['src/gmpy2.pxd']), ('gmpy2', ['src/gmpy2.h'])],
-      classifiers = [
+        if self.fast:
+            _comp_args.append('-DFAST=1')
+        if self.gcov:
+            if ON_WINDOWS:
+                raise ValueError("Cannot enable GCC code coverage on Windows")
+            _comp_args.append('-DGCOV=1')
+            _comp_args.append('-O0')
+            _comp_args.append('--coverage')
+            link_args.append('--coverage')
+            _libs.append('gcov')
+        if self.vector:
+            _comp_args.append('-DVECTOR=1')
+        if self.static and not ON_WINDOWS:
```

Comment:
Does this mean we can't have shared builds on windows?

Reply:
Yes, see below.

Reply:
This would break building conda packages.

Reply:
I think it would be best to not assume any defaults for windows and require `--static` for static builds. That will take away the 'on this platform this happens' scenarios.